src/Plugin/Field/FieldFormatter/EmbeddableFormatter.php

```diff
@@ -0,0 +1,176 @@
+ *
+ * @FieldFormatter (
+ *   id = "embeddable_formatter",
+ *   label = @Translation("Embeddable field formatter"),
+ *   description = @Translation("Field formatter for Embeddable media."),
+ *   field_types = {
+ *     "link",
+ *     "string",
+ *     "string_long",
+ *   }
+ * )
+ */
+class EmbeddableFormatter extends OEmbedFormatter {
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function isApplicable(FieldDefinitionInterface $field_definition) {
+    if ($field_definition->getTargetEntityTypeId() !== 'media' || !$field_definition->getTargetBundle()) {
+      return FALSE;
+    }
+
+    $media_type = MediaType::load($field_definition->getTargetBundle());
+    return $media_type && $media_type->getSource() instanceof Embeddable;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $embed_type = $items->getName();
+
+    if ($embed_type == "field_media_embeddable_oembed") {
+      // Here, we override the Drupal\media\Plugin\Field\FieldFormatter\oEmbedFormatter
+      // viewElements function, to make sure it works precisely the way we want.
+      // To use the original version in core, just uncomment the following line.
+      // return parent::viewElements($items, $langcode);.
+      $element = [];
+      $max_width = $this->getSetting('max_width');
+      $max_height = $this->getSetting('max_height');
+
+      foreach ($items as $delta => $item) {
+        $main_property = $item->getFieldDefinition()->getFieldStorageDefinition()->getMainPropertyName();
+        $value = $item->{$main_property};
+
+        if (empty($value)) {
+          continue;
+        }
+
+        try {
+          $resource_url = $this->urlResolver->getResourceUrl($value, $max_width, $max_height);
+          $resource = $this->resourceFetcher->fetchResource($resource_url);
+        }
+        catch (ResourceException $exception) {
+          $this->logger->error("Could not retrieve the remote URL (@url).", ['@url' => $value]);
+          continue;
+        }
+
+        if ($resource->getType() === Resource::TYPE_LINK) {
+          $element[$delta] = [
+            '#title' => $resource->getTitle(),
+            '#type' => 'link',
+            '#url' => Url::fromUri($value),
+          ];
+        }
+        elseif ($resource->getType() === Resource::TYPE_PHOTO) {
+          $element[$delta] = [
+            '#theme' => 'image',
+            '#uri' => $resource->getUrl()->toString(),
+            '#width' => $max_width ?: $resource->getWidth(),
+            '#height' => $max_height ?: $resource->getHeight(),
+          ];
+        }
+        else {
+          $url = Url::fromRoute('media.oembed_iframe', [], [
+            'query' => [
+              'url' => $value,
+              'max_width' => $max_width,
+              'max_height' => $max_height,
+              'hash' => $this->iFrameUrlHelper->getHash($value, $max_width, $max_height),
+            ],
+          ]);
```

Comment:
So, all that is copy-pasted from the OEmbedFormatter class, and it's included here because this function is an override.  The only stuff that's different here from the parent class is the logic for the iframe embeds below.

Should I refactor sections of code that's pulled directly from core?

Reply:
I would love to, but again, there are changes here to correct some problems with the core implementation.  For instance, in `parent::viewElements()` it adds a height and width attribute to the iframe tag, and that's an a11y error, per Caryl.  So I modified it here to add that stuff as CSS, and to add a few extra pixels to correct for when the iframe heights come in wrong, as was happening originally with Twitter.